### ISSUE TITLE
Horizontal scroll issue with TimelineHeaders

### ISSUE THREAD

**Issue by GNarek** [namespace-ee/react-calendar-timeline#809]
**Describe the bug**
When we are adding the `visibleTimeStart` and `visibleTimeEnd` attributes the header stops scrolling.
It doesn't work even if you add the `onTimeChange` with the function provided in examples:

**To Reproduce**
You can check it on your own example
[Controlled Scroll](https://github.com/namespace-ee/react-calendar-timeline/tree/master/examples#controlled-scroll)
[codesandbox Controlled Scroll](https://codesandbox.io/s/timeline-demo-controlled-visible-time-no-scroll-659jb)

**Expected behavior**
The header should be scrolled with the body

**Library Version**
0.27.0

**Desktop**

* OS: [MAC OS]
* Browser: [chrome]
* Version [Version 88.0.4324.146]

**Comment by cyclops26** [comment]
@GNarek I'm seeing the same issue. But I think it is expected - albeit missing some sort of error handling.

If I add `onTimeChange` to the timeline, the headers to move; however, I have significant jitter and make `visibleTimeStart` and `visibleTimeEnd` actually shift to different times then one I pass in.

The headers are based upon the visibleTimeStart and visibleTimeEnd versus the rows/items which are based on the Canvas.

Example based on visibleTimeStart and visibleTimeEnd being state properties:
```
/** will work **/
handleTimeChange = (visibleTimeStart, visibleTimeEnd) => {
  if (this._isMounted) this.setState({ visibleTimeStart, visibleTimeEnd });
}

/** will not work **/
handleTimeChange = (visibleTimeStart, visibleTimeEnd, updateScrollCanvas) => {
  updateScrollCanvas(visibleTimeStart, visibleTimeEnd)
}
```

It makes sense that `onTimeChange` is required because the Timeline needs a way for your `visibleTimeStart` and `visibleTimeEnd` properties to be updated (granted the code should probably enforce that requirement somehow instead of just appearing broken).

What I don't understand currently, at least in my code base, is why on initial Timeline render Timeline seems to be calling the `onTimeChange` function.

The jitter I'm seeing may just be a result of the `onTimeChange` calling a setState to save the new start/end times which in a scrolling scenario probably means it is probably calling setState too frequently and the constant re-render is messing with Timeline - so that is something I need to investigate further.


**Environment**
- OS: macOS - 11.2.1
- Browser: Chrome -> Version 89.0.4389.90 (Official Build) (x86_64)
- Browser: Firefox -> Version 85.0.2 (64-bit)
- Browser: Safari -> Version 14.0.3 (16610.4.3.1.4)

**Comment by GNarek** [comment]
@cyclops26 I don't think that the `onTimeChange` is required.

They can consider `visibleTimeStart` and `visibleTimeEnd` as `initialVisibleTimeStart` and  `initialVisibleTimeEnd`
and then calculate it by `timeSteps` and get the range instead of certain dates.

Yes using the `setState` in the scroll event will cause performance issues

**Comment by cyclops26** [comment]
@GNarek I agree - it would make much more sense if `visibleTimeStart`/`visibleTimeEnd` where initials - though I do wonder if they are not initials to allow people the ability to change the time window via an external call - though this could be resolved by implementing a way for an external process to call `updateScrollCanvas` or the equivalent hook into the `Timeline` component.

**Comment by cyclops26** [comment]
@GNarek I'm not sure how I missed it before, but there is `defaultTimeStart` and `defaultTimeEnd` that are initials - you can use that as your starting point and then handle the scrolling normally - that fixed my issue.
